src/inmanta/agent/executor.py

```diff
@@ -284,13 +326,32 @@ class VirtualEnvironmentManager:
     def __init__(self, envs_dir: str) -> None:
         self._environment_map: dict[EnvBlueprint, ExecutorVirtualEnvironment] = {}
         self.envs_dir: str = envs_dir
+        # We rely on a Named lock (`self._locks`) to be able to lock specific entries of the `_environment_map` dict. This
+        # allows us to prevent creating and deleting the same venv at a given time. The keys of this named lock are the hash of
+        # venv
         self._locks: NamedLock = NamedLock()
+        self._cleanup_scheduler = util.Scheduler("venv_cleanup_scheduler")
+
+    async def start(self) -> None:
+        interval = datetime.timedelta(days=1).total_seconds()
```

Comment:
What is this value? it is used as the `the period with which the executor touches its file!` and it is how often we do the cleanup check? 

Why one day?

Reply:
This is the interval of time for the cleanup check. It seems like a good balance if the default venv retention time is set to 7 days. It's not too aggressive